stockroom/repository.py

```diff
@@ -12,35 +85,38 @@ def init(name, email, overwrite):
     repo = Repository(Path.cwd(), exists=False)
     if repo.initialized and (not overwrite):
         commit_hash = repo.log(return_contents=True)['head']
-        print(f'Repo already exists at: {repo.path}')
+        print(f'Hangar Repo already exists at {repo.path}. '
+              f'Initializing it as stock repository')
     else:
+        if not all([name, email]):
+            raise ValueError("Both ``name`` and ``email`` has to be non-empty"
+                             " strings for initializing hangar repository")
         commit_hash = ''
         repo.init(user_name=name, user_email=email, remove_old=overwrite)
 
-    stock_file = Path.cwd().joinpath('head.stock')
+    stock_file = Path.cwd()/'head.stock'
```

Comment:
This is not how paths should ever be used in python. This will fail on windows systems (where the path seperator is `\` instead of `/`. Use the `pathlib.Path` class or `os.path.join` method for this to be automatically join paths with all separators handeld appropriately for you.

Reply:
This is not true after pathlib being part of python. Official doc uses slash to combine paths (https://docs.python.org/3.7/library/pathlib.html#operators) and pathlib takes care of operating system compatibility. Internally, It is essentially overloading the division operator

Reply:
Ahh my mistake! nice to know that's a thing :)